heclib/heclib_c/src/Public/zpdStore.c

```diff
@@ -274,9 +294,541 @@ int zpdStore(long long *ifltab, zStructPairedData *pds, int storageFlag)
+							for (int i = 0; i < pds->numberOrdinates; ++i) {
+								tmpFloatOrds[i] = pds->floatOrdinates[i] - offset;
+							}
+							origFloatOrds = pds->floatOrdinates;
+							pds->floatOrdinates = tmpFloatOrds;
+						}
+						else if (pds->doubleOrdinates) {
+							tmpDoubleOrds = (double *)calloc(pds->numberOrdinates, sizeof(double));
+							for (int i = 0; i < pds->numberOrdinates; ++i) {
+								tmpDoubleOrds[i] = pds->doubleOrdinates[i] - offset;
+							}
+							origDoubleOrds = pds->doubleOrdinates;
+							pds->doubleOrdinates = tmpDoubleOrds;
+						}
+					}
+				}
+			}
+			if (depElev) {
+				ivertical_datum = getEffectiveVerticalDatum(
+					cvertical_datum,
+					sizeof(cvertical_datum),
+					&pds->userHeader,
+					&pds->userHeaderNumber,
+					&pds->unitsDependent);
+				//-------------------------------------------------------//
+				// now that we have a datum, determine the offset to use //
+				//-------------------------------------------------------//
+				switch(ivertical_datum) {
+					case IVERTICAL_DATUM_NAVD88 :
+						offset = vdi->offsetToNavd88;
+						break;
+					case IVERTICAL_DATUM_NGVD29 :
+						offset = vdi->offsetToNgvd29;
+						break;
+					default :
+						if(!strcmp(cvertical_datum, vdi->nativeDatum) || !strcmp(cvertical_datum, CVERTICAL_DATUM_OTHER)) {
+							offset = 0;
+						}
+						else {
+							if (indElev) {
+								ivertical_datum = getEffectiveVerticalDatum(
+									cvertical_datum,
+									sizeof(cvertical_datum),
+									&headerCopy,
+									&headerCopyNumber,
+									&pds->unitsDependent);
+								switch(ivertical_datum) {
+									case IVERTICAL_DATUM_NAVD88 :
+										offset = vdi->offsetToNavd88;
+										break;
+									case IVERTICAL_DATUM_NGVD29 :
+										offset = vdi->offsetToNgvd29;
+										break;
+									default :
+										if(!strcmp(cvertical_datum, vdi->nativeDatum) || !strcmp(cvertical_datum, CVERTICAL_DATUM_OTHER)) {
+											offset = 0;
+										}
+										else {
+											offset = UNDEFINED_VERTICAL_DATUM_VALUE;
+										}
+										break;
+								}
+							}
+							else {
+								offset = UNDEFINED_VERTICAL_DATUM_VALUE;
+							}
+						}
+						break;
+				}
+				if (offset != 0.) {
+					char errmsg[256];
+					if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
+						sprintf(
+							errmsg,
+							"\nVertical datum offset is undefined for datum '%s'.\n"
+							"Datum conversion could not be performed.\n"
+							"No data stored.",
+							cvertical_datum);
+						if (vdiPd && vdiPd != &_vdiPd) {
+							free(vdiPd);
+						}
+						return zerrorProcessing(ifltab, DSS_FUNCTION_zpdStore_ID,
+							zdssErrorCodes.INCOMPATIBLE_CALL, 0,
+							0, zdssErrorSeverity.WARNING, pds->pathname,
+							errmsg);
+					}
+					offset = getOffset(offset, vdi->unit, indElev ? pds->unitsIndependent : pds->unitsDependent);
+					if (offset == UNDEFINED_VERTICAL_DATUM_VALUE) {
```

Comment:
The following if block is redundant. We handled indElev == TRUE above and are now in an if block where depElev == TRUE